lib/notifications/display.dart

```diff
@@ -395,4 +391,59 @@ class NotificationDisplayManager {
     }
     return null;
   }
+
+  /// Generates a uri from the FCM message data, embedding the [narrowLink]
+  /// and the userID.
+  ///
+  /// This uri is used to route to the relevant conversation's [MessageListPage]
+  /// when a notification with this intent uri is opened.
+  @visibleForTesting
+  static Future<Uri?> notificationIntentUriFromFcmMessage({
+    required GlobalStore globalStore,
+    required MessageFcmMessage data,
+  }) async {
+    final account = globalStore.accounts.firstWhereOrNull((account) =>
+      account.realmUrl.origin == data.realmUri.origin && account.userId == data.userId);
```

Comment:
notif: Match realmUri using only the origin part of the uri

Can you elaborate on the motivation for this change? What's the situation where it makes a difference, and why is the new behavior better than the old behavior in that situation?

Reply:
Added some reasoning in the commit message, but mainly it's because when re-creating the realmUri from narrowUri (using `Uri.parse(narrowUri.origin)`), we don't get the trailing `/` which is present in `account.realmUri`, resulting in matching to fail.

Reply:
Removed this change because `narrowUri` doesn't exist anymore, thus this change is not required anymore. Will send a separate PR to fix `example_data.dart`, with corresponding tests.